pkg/sdk/sdk_test.go

```diff
@@ -0,0 +1,503 @@
+					vars := mux.Vars(r)
+					id, ok := vars["id"]
+					if !ok {
+						response.ToJson(w, http.StatusBadRequest, "id not found", nil)
+						return
+					}
+					err := r.ParseMultipartForm(10000)
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file size exceeded", nil)
+						log.Error(err.Error())
+						return
+					}
+					file, _, err := r.FormFile("file")
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file parse fail", nil)
+						log.Error(err.Error())
+						return
+					}
+					fileBytes, err := ioutil.ReadAll(file)
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file read fail", nil)
+						log.Error(err.Error())
+						return
+					}
+					if !reflect.DeepEqual(id, "1") {
+						t.Errorf("Want: %v, Got: %v", "1", id)
+					}
+					if !reflect.DeepEqual(fileBytes, []byte("abc")) {
+						t.Errorf("Want: %v, Got: %v", []byte("abc"), fileBytes)
+					}
+					defer file.Close()
+					cType := r.Header.Get("Content-Type")
+					x := strings.Split(cType, ";")
+					if !reflect.DeepEqual(x[0], "multipart/form-data") {
+						t.Errorf("Want: %v, Got: %v", "multipart/form-data", x[0])
+					}
+					response.ToJson(w, http.StatusOK, "SUCCESS", map[string]interface{}{
+						"id": id,
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: type inference fail",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusOK, "SUCCESS", Reader(""))
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "unable to parse response data" {
+					t.Errorf("Want: %v, Got: %v", "unable to parse response data", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: json failure",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+
+					json.NewEncoder(w).Encode("")
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "json: cannot unmarshal string into Go value of type model.Response" {
+					t.Errorf("Want: %v, Got: %v", "json: cannot unmarshal string into Go value of type model.Response", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: incorrect id in response",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					vars := mux.Vars(r)
```

Comment:
If the parsed data is not required, remove the logic to parse the file data.
Looks like currently, it is reading the data using FormFile() and closing it without doing anything.

This being a failure case, only put the required logic to passing a non-matching `id`.

Reply:
done